src/Controllers/GagePageController.ts

```diff
@@ -779,10 +847,41 @@ module StreamStats.Controllers {
                     // console.log(error)
                 }).finally(() => {
                     this.formatData()
+                    this.updateChart()
+                    this.getMinYear()
                 });
-
         } 
 
+        public getMinYear() {
+            const minYear = Math.min.apply(
+              null, 
+              this.measuredObj.map((item) => {
+                const itemDate = new Date(item.dateTime);
+                return itemDate.getFullYear();
+              })
+            );
+            return minYear;
+          }
+      
+        public updateChart() {
+            console.log('measured obj',this.measuredObj )
+            let chart = $('#chart3').highcharts();
+            chart.series[2].update({data:[]});
+            const filteredData = this.measuredObj.filter((item) => {
+                const itemDate = new Date(item.dateTime);
+                const itemMonth = itemDate.getMonth() + 1;
+                const itemYear = itemDate.getFullYear();
+                return itemMonth >= this.startMonth && itemMonth <= this.endMonth &&
+                itemYear >= this.startYear && itemYear <= this.endYear;
+            });
+            console.log('filtered data',filteredData )
+        
+            if (chart) {
+                console.log('test', chart)
+                chart.series[2].update({data:filteredData});
+            }
+        }
+        
```

Comment:
These updates will allow both the date sliders and age/quality radio buttons to be handled in a single function and work correctly

Then you can delete the separate `toggleDischargeData` function.

```suggestion
            let chart = $('#chart3').highcharts();
            // Move everything else to be inside the if(chart)
            if(chart) {
                chart.series[2].update({data:[]}); // Reset the data series 

                // Start with a fresh copy of the original data series (this.measuredObj)
                // To avoid affecting the data associated with the data series (this.measuredObj), create a "deep copy" of this.measuredObj with the structuredClone function
                // Filter the data based on selected dates
                const filteredData = structuredClone(this.measuredObj).filter((item) => {
                    const itemDate = new Date(item.dateTime);
                    const itemMonth = itemDate.getMonth() + 1;
                    const itemYear = itemDate.getFullYear();
                    return itemMonth >= this.startMonth && itemMonth <= this.endMonth &&
                    itemYear >= this.startYear && itemYear <= this.endYear;
                });

                // Assign the correct color based on age or quality selection
                filteredData.forEach(row => {
                    row.color = (this.ageQualityData == 'age') ? row.ageColor : row.qualityColor;
                });

                // Update the data for this series
                chart.series[2].update({data:filteredData});
            }
        
```